sdk/eventhub/event-hubs/src/util/parseEndpoint.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Parses the host, hostname, and port from an endpoint.
+ * @param endpoint And endpoint to parse.
+ * @hidden
+ * @internal
+ */
+export function parseEndpoint(endpoint: string): { host: string; hostname: string; port?: string } {
+  const hostMatch = endpoint.match(/.*:\/\/([^\/]*)/i);
```

Comment:
You can drop the `/i` (ie, you have no text in your regex to be case-insensitive against).

Also, did we establish you can transform `[^\/]` into `[^/]`?

Reply:
It can indeed!

Reply:
btw, why can't we use URL here to parse apart the string?

Reply:
URL fails for certain schemes. I tried that originally, but if you attempted to connect to something where the scheme was `sb://` or `amqp://` then it failed to actually parse the url into its constituent parts.